frontend/linux/distro_handler.go

```diff
@@ -0,0 +1,177 @@
+package linux
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/containerd/platforms"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/client/llb/sourceresolver"
+	"github.com/pkg/errors"
+
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type DistroConfig interface {
```

Comment:
The idea here is to have an interface that would work across distros, even beyond rpm/deb based packaging. Right now though, it just conforms to what is needed for rpm and deb. There are a few things that don't fit perfectly here, in particular the need for an extra `worker` parameter for building an rpm container when it isn't needed for deb. The signatures are similar enough though that it seemed like an interface would be reasonable, but I'm definitely open to rethinking this if it's too much of a stretch.

Reply:
I'll say the large interface makes me a little hesitant but it doesn't seem too bad here.
If we need to add something that can't conform to it then it'll just need to handle some of these higher level functions on their own.

Basically, SGTM.